_drafts/_draft-2024-02-22-connectivity-metrics.md

```diff
@@ -0,0 +1,232 @@
+
+<p align="center">
+  <img width="650" src="{% img_url connectivity-metrics/wireshark-ble-capture.png %}" alt="BLE Packet Capture in Wireshark" />
+</p>
+
+For this problem, I observed lower throughput than expected during data transfer. Logs showed that the connection parameters were set correctly initially, but the data rate did not match this. Inspecting the capture in Wireshark led me to discover that two different connection parameter update methods in the BLE stack clobbered values set by the other, resulting in decreased throughput.
+
+Depending on the protocol, you may not need additional hardware. Wireshark can use your WiFi interface’s monitor or promiscuous mode to capture any signals in the air. It can also hook into your local Ethernet and USB interfaces for packet capture. For BLE sniffing hardware, I reach for my closest nRF52 dev kit! Nordic Semiconductor offers a BLE Sniffer package to download that includes plugins for Wireshark. With these, Wireshark will use your dev kit as an interface to capture packets from.
+
+**Strengths**:
+
+- Incredibly detailed analysis of packets on the wire/over the air
+- Great for investigating specific protocol issues
+
+**Best Used For**:
+
+- Single devices
+- Local debugging/investigation
+- Protocol errors and profiling specific devices
+
+Use protocol analyzers when you need extreme detail for a **single** device at your bench or desk. These tools will not help in the field unless you’re present. They should rarely be used because packet captures are very large and process-intense. Packet captures focus on a single device and can’t be used to aggregate across a growing fleet.
+
+### Attributes
+
+Attributes are used to supplement logging, metrics, and protocol traces. Attributes are bits of additional info related to the device that change infrequently or not at all. These attributes provide more context as to what is unique about the device. Some examples are component vendors, modem firmware versions, gateway/phone OS versions, or geography identifiers. An attribute could help identify a particular battery provider between devices that frequently lose connection due to out-of-spec power output.
```

Comment:
Add example screenshot and why it's useful

Reply:
Agree that an example here would beef up this section (shapemate screenshot perhaps with the production version attribute?)